geode-core/src/test/java/org/apache/geode/cache/query/internal/types/TypeUtilsJUnitTest.java

```diff
@@ -0,0 +1,1197 @@
+    // We use spies to track the execution of wanted and unwanted methods.
+    PowerMockito.spy(TypeUtils.class);
+    NumericComparator numericComparator = spy(NumericComparator.class);
+    TemporalComparator temporalComparator = spy(TemporalComparator.class);
+    PowerMockito.when(TypeUtils.getNumericComparator()).thenReturn(numericComparator);
+    PowerMockito.when(TypeUtils.getTemporalComparator()).thenReturn(temporalComparator);
+
+    // Spies to make sure that other comparison methods are not executed.
+    Date beginningOfTimeAsDate = spy(new Date(0L));
+    Date currentCalendarTimeAsDate = spy(Calendar.getInstance().getTime());
+    java.sql.Date beginningOfTimeAsSqlDate =
+        spy(new java.sql.Date(beginningOfTimeAsDate.getTime()));
+    java.sql.Date currentCalendarTimeAsSqlDate =
+        spy(new java.sql.Date(currentCalendarTimeAsDate.getTime()));
+    java.sql.Time beginningOfTimeAsSqlTime =
+        spy(new java.sql.Time(beginningOfTimeAsDate.getTime()));
+    java.sql.Time currentCalendarTimeAsSqlTime =
+        spy(new java.sql.Time(currentCalendarTimeAsDate.getTime()));
+    java.sql.Timestamp beginningOfTimeAsSqlTimestamp =
+        spy(new java.sql.Timestamp(beginningOfTimeAsDate.getTime()));
+    java.sql.Timestamp currentCalendarTimeAsSqlTimestamp =
+        spy(new java.sql.Timestamp(currentCalendarTimeAsDate.getTime()));
+
+    List<Object> originDates = Arrays.asList(new Object[] {beginningOfTimeAsDate,
+        beginningOfTimeAsSqlDate, beginningOfTimeAsSqlTime, beginningOfTimeAsSqlTimestamp});
+    List<Object> currentDates =
+        Arrays.asList(new Object[] {currentCalendarTimeAsDate, currentCalendarTimeAsSqlDate,
+            currentCalendarTimeAsSqlTime, currentCalendarTimeAsSqlTimestamp});
+
+    originDates.forEach(originDate -> {
+      originDates.forEach(originDate2 -> {
+        try {
+          assertThat(TypeUtils.compare(originDate, originDate2, OQLLexerTokenTypes.TOK_EQ))
+              .isEqualTo(Boolean.TRUE);
+          assertThat(TypeUtils.compare(originDate, originDate2, OQLLexerTokenTypes.TOK_LT))
+              .isEqualTo(Boolean.FALSE);
+          assertThat(TypeUtils.compare(originDate, originDate2, OQLLexerTokenTypes.TOK_LE))
+              .isEqualTo(Boolean.TRUE);
+          assertThat(TypeUtils.compare(originDate, originDate2, OQLLexerTokenTypes.TOK_GT))
+              .isEqualTo(Boolean.FALSE);
+          assertThat(TypeUtils.compare(originDate, originDate2, OQLLexerTokenTypes.TOK_GE))
+              .isEqualTo(Boolean.TRUE);
+          assertThat(TypeUtils.compare(originDate, originDate2, OQLLexerTokenTypes.TOK_NE))
+              .isEqualTo(Boolean.FALSE);
+          verify((Comparable) originDate, times(0)).compareTo(any());
+          verify((Comparable) originDate2, times(0)).compareTo(any());
+          verify(temporalComparator, times(6)).compare(any(), any());
+          reset(temporalComparator);
+        } catch (TypeMismatchException typeMismatchException) {
```

Comment:
Remove any try-catch blocks and simply add "throws Exception" to the signature of the test methods.

Reply:
Hey @kirklund 

Thanks for reviewing this. By default Java doesn't support throwing checked exceptions from a lambda expression, that's why I'm catching the checked exceptions and re-throwing them as unchecked exceptions in the tests. There are certain ways to [handle checked exceptions in Java streams](https://www.oreilly.com/ideas/handling-checked-exceptions-in-java-streams), but they're not trivial, and not even worth the hassle for test classes. If an exception is thrown the test will fail anyway, and the root cause of the exception will be shown as for every other test.
Let me know if you still want this to be changed.
Cheers.

Reply:
Oops, you're right! I was thinking these were dunit vm invocations which you can remove the catch-blocks from lambda calls.

Reply:
👍